File: CodingSamples/DotNet/Fundamentals/Fundamentals/Inheritance/ClassTypeTest/DemoApp1/Payroll/SalesPerson.cs
namespace Payroll
{
    //SalesPerson is a derived class of Employee(base class)
    class SalesPerson : Employee
    {
        //automatic property
        public double Sales { get; set; }

        //derived class constructor must call base class constructor
        //otherwise parameterless constructor of base class is called
        //implicitly
        public SalesPerson(int h, float r, double s) : base(h, r)
        {
            Sales = s;
        }

        //method overriding - is defining a method (with override keyword)
        //in derived class whose declaration matches with the declaration
        //of a virtual method in the base class
        public override double Income()
        {
            double payment = base.Income();
            if(Sales >= 25000)
                payment += 0.05 * Sales;
            return payment;
        }
    }
}

File: CodingSamples/Java/Fundamentals/Inheritance/SubclassTest/demo1/payroll/SalesPerson.java
package payroll;

//defining SalesPerson as a subclass of Employee
public class SalesPerson extends Employee {
    
    private double sales;

    public SalesPerson(int h, float r, double s) {
        super(h, r);
        sales = s;
    }

    public double getSales() {
        return sales;
    }

    public void setSales(double value) {
        sales = value;
    }

    //overriding method of superclass by defining a method with 
    //a matching declaration in the subclass
    public double income() {
        double payment = super.income();
        if(sales >= 25000)
            payment += 0.05 * sales;
        return payment;
    }
}

